lib/pure/streams.nim

```diff
@@ -181,6 +233,30 @@ proc readLine*(s: Stream, line: var TaintedString): bool =
     line.string.add(c)
   result = true
 
+proc peekLine*(s: Stream, line: var TaintedString): bool =
+  ## peeks a line of text from the stream `s` into `line`. `line` must not be
+  ## ``nil``! May throw an IO exception.
+  ## A line of text may be delimited by ``CR``, ``LF`` or
+  ## ``CRLF``. The newline character(s) are not part of the returned string.
+  ## Returns ``false`` if the end of the file has been reached, ``true``
+  ## otherwise. If ``false`` is returned `line` contains no new data.
+  line.string.setLen(0)
+  let pos = getPosition(s)
+  while true:
+    var c = readChar(s)
+    if c == '\c': 
+      c = readChar(s)
+      break
+    elif c == '\L': break
+    elif c == '\0':
+      if line.len > 0: break
+      else: 
+        setPosition(s, pos)
+        return false
+    line.string.add(c)
+  setPosition(s, pos)
```

Comment:
Might be a good idea to put this after a ``defer`` (http://nim-lang.org/docs/manual.html#exception-handling-defer-statement).